.github/workflows/clang-format-check.yml

```diff
@@ -10,4 +10,4 @@ jobs:
       uses: jidicula/clang-format-action@v4.0.0
       with:
         clang-format-version: '12'
-        exclude-regex: '(build|config|deps)'
+        exclude-regex: '(build|config|deps|base64)'
```

Comment:
Might the `base64` library live in the `deps/` tree, instead of directly in the `include/` path? It would be cleaner to keep all third-party code in the same place. It would also be pretty easy to wrap the library as its own CMake target, which would make the build system code a little cleaner. I'm happy to write these changes up as a PR to this branch if you like.

Reply:
Yeah, if you think that's appropriate, go ahead. This seemed like the simplest way to me, but if you want to go through that process, go ahead and I'll merge it into this branch.